evio/src/THaEtClient.C

```diff
@@ -240,6 +247,129 @@ Int_t THaEtClient::codaRead()
   return status;
 }
 
+//______________________________________________________________________________
+Int_t THaEtClient::codaWrite( const UInt_t* buffer, UInt_t buffer_length )
+{
+  if( !opened ) {
+    Int_t status = init(station.c_str());
+    if( status != CODA_OK ) {
+      cout << "THaEtClient: ERROR: codaWrite, cannot connect to CODA" << endl;
+      return CODA_FATAL;
+    }
+  }
+
+  // Get a new chunk of ET events if we've used all current ones
+  if( evh.currentChunkID >= evh.etChunkSize || evh.currentChunkID < 0 ) {
+    // Need enough space for event data plus EVIO overhead
+    // EVIO v4 block header = 32 bytes (EV_HDSIZ_BYTES)
+    // EVIO v6 block header = 56 bytes (EV_HDSIZ_BYTES_V6)
+    // See evio/src/libsrc/evio.h for EV_HDSIZ_BYTES(_V6) definitions
+#ifdef EV_HDSIZ_V6
+    constexpr size_t EV_HDSIZ_BYTES = st::max(4 * EV_HDSIZ, 4 * EV_HDSIZ_V6);
```

Comment:
Need to use different variable names here. These will be replaced by new defines in evio v6.